src/lib/storage/proposal/devicegraph_generator.rb

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2016] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "storage/proposal/refined_devicegraph"
+require "storage/proposal/space_maker"
+require "storage/proposal/partition_creator"
+
+module Yast
+  module Storage
+    module Proposal
+      using RefinedDevicegraph
+
+      # Class to create devicegraphs that can accommodate a given collection of
+      # volumes
+      class DevicegraphGenerator
```

Comment:
Why throw away the speaking name StorageProposal and replace it with a nondescript name DevicegraphGenerator?

We are not in the business of making or collecting device graphs. This should be about making a useful storage proposal, and I don't see that name any more at all. This is a poor design choice. Whoever is reading this code will search forever to find anything that matches what he sees on the screen during installation, or well-known concepts in the YaST code.

I completely disagree with this choice.